# Assignment 4
### Name: Muhammad Zeeshan
### Email: zeeshanayaz1@gmail.com
### Roll: 334256

## Task1:
You need to write a function in python that return all such numbers which are divisible by 7 but are not a multiple of 5, between 2000 and 3200 (both included).
- Return the obtained numbers in a comma-separated sequence on a single line. 

In [2]:
def print_seven_divisible():
    for i in range(2000, 3200):
        if (i % 7 == 0) and (i % 5 != 0):
            print(f'{i}', end=',')

In [3]:
print_seven_divisible()

2002,2009,2016,2023,2037,2044,2051,2058,2072,2079,2086,2093,2107,2114,2121,2128,2142,2149,2156,2163,2177,2184,2191,2198,2212,2219,2226,2233,2247,2254,2261,2268,2282,2289,2296,2303,2317,2324,2331,2338,2352,2359,2366,2373,2387,2394,2401,2408,2422,2429,2436,2443,2457,2464,2471,2478,2492,2499,2506,2513,2527,2534,2541,2548,2562,2569,2576,2583,2597,2604,2611,2618,2632,2639,2646,2653,2667,2674,2681,2688,2702,2709,2716,2723,2737,2744,2751,2758,2772,2779,2786,2793,2807,2814,2821,2828,2842,2849,2856,2863,2877,2884,2891,2898,2912,2919,2926,2933,2947,2954,2961,2968,2982,2989,2996,3003,3017,3024,3031,3038,3052,3059,3066,3073,3087,3094,3101,3108,3122,3129,3136,3143,3157,3164,3171,3178,3192,3199,

## Task2:
You need code a function that calculates and returns the value according to the given formula:
- P = Square root of [(2 * A * B)/C]
Following are the fixed values of A and B:
- A is 50. B is 30.
The values of the literal C should be taken as console input to your program in a comma-separated sequence.
- Example
    - Let us assume the following comma separated input sequence is given to the program: 100,150,180
    - The output of the program should be: 18,22,24

In [15]:
import math

def calculate(c, a=50, b=30):
    if c == 0:
        return "Division by zero is not allowed."
    result = math.sqrt((2 * a * b) / c)
    return round(result)

In [16]:
numbers = input('Enter three number for c seperated by qoma: ')
number_list = numbers.split(',')
for number in number_list:
    print(f'{str(calculate(int(number)))}')

Enter three number for c seperated by qoma:  100,150,180


5
4
4


## Task3:
You need to write a function that takes a comma separated sequence of words as input and prints the words in a comma-separated sequence after sorting them alphabetically.
- Suppose the following input is supplied to the program: without,hello,bag,world
- Then, the output should be: bag,hello,without,world 


In [17]:
def sort_words(input_string):
    words = input_string.split(',')
    
    words.sort()
    
    return ','.join(words)

In [18]:
input_string = input("Enter a comma-separated sequence of words: ")
sorted_words = sort_words(input_string)
print("Sorted words:", sorted_words)

Enter a comma-separated sequence of words:  without,hello,bag,world


Sorted words: bag,hello,without,world


## Task 3
You need to write a program that takes sequence of lines as input and prints the lines after making all characters in the sentence capitalized.
- Suppose the following input is supplied to the program:
    - Hello world
    - Practice makes perfect
- Then, the output should be:
    - HELLO WORLD
    - PRACTICE MAKES PERFECT 

In [19]:
def capitalize_lines():
    # Take multiple lines of input from the user
    while True:
        try:
            line = input()
            if line == '':
                break
            # Print the line in uppercase
            print(line.upper())
        except EOFError:
            break

In [20]:
print("Enter multiple lines. Press Enter on an empty line to stop:")
capitalize_lines()

Enter multiple lines. Press Enter on an empty line to stop:


 Hello world


HELLO WORLD


 Practice makes perfect


PRACTICE MAKES PERFECT


## Task5:
You need to write a function that counts the number of vowels in a given sentence as input from console.
- Suppose the following input is supplied to the program:
    - Practice makes perfect
Then, the output should be:
  - a appeared 2 times
  - e appeared 5 times
  - i appeared 1 times
  - o appeared 2 times
  - u appeared 0 time
  - PRACTICE MAKES PERFECT

In [1]:
def count_vowels(data):
    vowel_list = ['a','e','i','o','u']
    vowel_counts = [0,0,0,0,0]
    
    for char in data:
        if char.lower() in vowel_list:
            index = vowel_list.index(char.lower())
            vowel_counts[index]+=1
    return vowel_list, vowel_counts

In [4]:
count_vowels('Hello world Practice makes perfect')

(['a', 'e', 'i', 'o', 'u'], [2, 5, 1, 2, 0])

In [5]:
def count_vowels_in_lines():
    vowel_list = ['a','e','i','o','u']
    vowel_counts = [0,0,0,0,0]
    
    while True:
        try:
            line = input()
            if line == '':
                return vowel_list, vowel_counts
            for char in line:
                if char.lower() in vowel_list:
                    index = vowel_list.index(char.lower())
                    vowel_counts[index]+=1
        except EOFError:
            break

In [7]:
vowels, vowels_count = count_vowels_in_lines()
for index, vowel in enumerate(vowels):
    print(f'{vowel} appeared {vowels_count[index]} times')

 hello world
 practice makes perfect
 


a appeared 2 times
e appeared 5 times
i appeared 1 times
o appeared 2 times
u appeared 0 times


## Task6:
You need write a function that traces and makes a list of all such numbers from1000 to 3000 in which all the digits are even numbers.

In [11]:
def even_numbers():
    even_numbers = []
    for number in range(1000,3000+1):
        if number % 2 == 0:
            even_numbers.append(number)
    return even_numbers

In [12]:
even_numbers()

[1000,
 1002,
 1004,
 1006,
 1008,
 1010,
 1012,
 1014,
 1016,
 1018,
 1020,
 1022,
 1024,
 1026,
 1028,
 1030,
 1032,
 1034,
 1036,
 1038,
 1040,
 1042,
 1044,
 1046,
 1048,
 1050,
 1052,
 1054,
 1056,
 1058,
 1060,
 1062,
 1064,
 1066,
 1068,
 1070,
 1072,
 1074,
 1076,
 1078,
 1080,
 1082,
 1084,
 1086,
 1088,
 1090,
 1092,
 1094,
 1096,
 1098,
 1100,
 1102,
 1104,
 1106,
 1108,
 1110,
 1112,
 1114,
 1116,
 1118,
 1120,
 1122,
 1124,
 1126,
 1128,
 1130,
 1132,
 1134,
 1136,
 1138,
 1140,
 1142,
 1144,
 1146,
 1148,
 1150,
 1152,
 1154,
 1156,
 1158,
 1160,
 1162,
 1164,
 1166,
 1168,
 1170,
 1172,
 1174,
 1176,
 1178,
 1180,
 1182,
 1184,
 1186,
 1188,
 1190,
 1192,
 1194,
 1196,
 1198,
 1200,
 1202,
 1204,
 1206,
 1208,
 1210,
 1212,
 1214,
 1216,
 1218,
 1220,
 1222,
 1224,
 1226,
 1228,
 1230,
 1232,
 1234,
 1236,
 1238,
 1240,
 1242,
 1244,
 1246,
 1248,
 1250,
 1252,
 1254,
 1256,
 1258,
 1260,
 1262,
 1264,
 1266,
 1268,
 1270,
 1272,
 1274,
 1276,
 1278,
 1280,
 1282,
 1284,

## Task7:
You need to write a code which accepts a sequence of comma separated 4 digit binary numbers as its input and then check whether they are divisible by 5 or not.
The numbers that are divisible by 5 are to be printed in a comma separated
sequence.
Example:
0100,0011,1010,1001
Then the output should be:
1010
Reference: https://www.datacamp.com/.../python-data-type-conversion

In [17]:
def check_divisible_by_5(binary_numbers):
    divisible_by_5 = []
    for binary in binary_numbers:
        decimal = int(binary, 2)
        if decimal % 5 == 0:
            divisible_by_5.append(binary)
    return ','.join(divisible_by_5)

In [18]:
input_numbers = input("Enter comma-separated 4-digit binary numbers: ")
binary_numbers = input_numbers.split(',')
divisible_by_5 = check_divisible_by_5(binary_numbers)
print(divisible_by_5)

Enter comma-separated 4-digit binary numbers:  0100,0011,1010,1001


1010


## Task8:
Write a program that accepts a sentence and calculate the number of letters and digits.
- Suppose the following input is supplied to the program:
    - hello world! 123
- Then, the output should be:
    - LETTERS 10
    - DIGITS 3 

In [15]:
def calculate_letter_count(data):
    letter_count = 0
    digit_count = 0
    for char in data:
        if char.isalpha():
            letter_count+=1
        elif char.isdigit():
            digit_count+=1
    return letter_count, digit_count

In [16]:
sentence = input('Enter sentence to count letter and digit: ')
letter_count, digit_count = calculate_letter_count(sentence)
print(f'Letters {letter_count}')
print(f'Digit {digit_count}')

Enter sentence to count letter and digit:  hello world! 123


Letters 10
Digit 3


## End